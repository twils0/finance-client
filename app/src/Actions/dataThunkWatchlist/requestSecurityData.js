import axios from 'axios';
import raven from 'raven-js';

import { requestStatusTypes, axiosConfig, URLs } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsWatchlist';
import { setWatchlistStatus, setSecurityData } from '../dataActionsWatchlist';

import handleErrorCatch from '../../handleErrorCatch';
import convertPrice from '../../Functions/convertPrice';
import convertDate from '../../Functions/convertDate';
import convertTime from '../../Functions/convertTime';

const requestSecurityData = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'id')) {
    throw new Error(`Please enter a value for 'id' - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'tickerCusip')) {
    throw new Error(`Please enter a value for 'tickerCusip' - ${JSON.stringify(payload)}`);
  }

  return async (dispatch, getState) => {
    const state = getState();

    if (
      state.data.watchlist.status[statusNames.GET_SECURITY_DATA].status
      !== requestStatusTypes.LOADING
    ) {
      dispatch(
        setWatchlistStatus({
          id: statusNames.GET_SECURITY_DATA,
          status: requestStatusTypes.LOADING,
        }),
      );

      const { id, tickerCusip } = payload;
      let data = null;

      try {
        const [responseCompany, responseQuote] = await Promise.all([
          axios.get(URLs.IEX_COMPANY(tickerCusip), {
            params: {
              filter: 'description',
            },
            ...axiosConfig.IEX,
          }),
          axios.get(URLs.IEX_QUOTE(tickerCusip), {
            params: {
              filter: 'latestPrice,latestUpdate',
            },
            ...axiosConfig.IEX,
          }),
        ]);

        const description = responseCompany.data.description.replace('\n ', '\n\n');

        const { latestPrice, latestUpdate } = responseQuote.data;

        data = {
          description,
          price: {
            amount: convertPrice(latestPrice),
            date: convertDate(latestUpdate),
            time: convertTime(latestUpdate),
          },
        };
      } catch (errorCatch) {
        const error = handleErrorCatch(errorCatch);

        dispatch(
          setWatchlistStatus({
            id: statusNames.GET_SECURITY_DATA,
            status: requestStatusTypes.ERROR,
          }),
        );

        raven.captureException(error, {
          logger: 'requestSecurityData',
        });

        return Promise.reject(error);
      }

      dispatch(setSecurityData({ id, data }));

      dispatch(
        setWatchlistStatus({
          id: statusNames.GET_SECURITY_DATA,
          status: requestStatusTypes.SUCCESS,
        }),
      );
    }

    return null;
  };
};

export default requestSecurityData;
