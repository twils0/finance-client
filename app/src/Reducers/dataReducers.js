import { combineReducers } from 'redux';

import auth from './dataReducersAuth';
import aws from './dataReducersAWS';
import watchlist from './dataReducersWatchlist';
import account from './dataReducersAccount';

const dataReducers = combineReducers({
  auth,
  aws,
  watchlist,
  account,
});

export default dataReducers;
