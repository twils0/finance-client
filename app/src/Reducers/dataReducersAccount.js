import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames, fieldNames } from '../Constants/dataConstantsAccount';

export const initialState = {
  status: {
    [statusNames.AWS_FIELDS]: {
      id: statusNames.AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.STRIPE_FIELDS]: {
      id: statusNames.STRIPE_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_TOKEN]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_TOKEN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_REQUEST]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_REQUEST,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_DB_FIELDS]: {
      id: statusNames.UPDATE_DB_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_AWS_FIELDS]: {
      id: statusNames.UPDATE_AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
  },
  fields: {
    list: [...Object.values(fieldNames)],
    [fieldNames.NAME]: {
      id: fieldNames.NAME,
      value: '',
    },
    [fieldNames.EMAIL]: {
      id: fieldNames.EMAIL,
      value: '',
    },
    [fieldNames.EMAIL_ADDITIONAL]: {
      id: fieldNames.EMAIL_ADDITIONAL,
      value: '',
    },
    [fieldNames.PHONE]: {
      id: fieldNames.PHONE,
      value: '',
    },
    [fieldNames.NAME_ON_CARD]: {
      id: fieldNames.NAME_ON_CARD,
      value: '',
    },
    [fieldNames.PROMO_CODE]: {
      id: fieldNames.PROMO_CODE,
      value: '',
    },
    [fieldNames.PROMO_CODE_VALID]: {
      id: fieldNames.PROMO_CODE_VALID,
      value: true,
    },
    [fieldNames.CUSTOMER]: {
      id: fieldNames.CUSTOMER,
      value: '',
    },
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_ACCOUNT_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const fields = (state = initialState.fields, action) => {
  switch (action.type) {
    case actionTypes.SET_FIELD:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_FIELDS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.fields;
    default:
      return state;
  }
};

const dataReducersAccount = combineReducers({
  status,
  fields,
});

export default dataReducersAccount;
