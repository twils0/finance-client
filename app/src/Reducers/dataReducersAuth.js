import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
