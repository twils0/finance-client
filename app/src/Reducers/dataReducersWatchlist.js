import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames } from '../Constants/dataConstantsWatchlist';

export const initialState = {
  status: {
    [statusNames.SECURITIES]: {
      id: statusNames.SECURITIES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.GET_SECURITY_DATA]: {
      id: statusNames.GET_SECURITY_DATA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_SECURITIES]: {
      id: statusNames.UPDATE_SECURITIES,
      status: requestStatusTypes.IDLE,
    },
  },
  securities: {
    current: null,
    list: [],
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_WATCHLIST_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const securities = (state = initialState.securities, action) => {
  switch (action.type) {
    case actionTypes.SET_SECURITIES_CURRENT:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_LIST:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_ALL:
      return {
        ...action.payload.securities,
      };
    case actionTypes.SET_SECURITY:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_SECURITY_DATA: {
      if (!state[action.payload.id]) {
        return {
          ...state,
          [action.payload.id]: {
            ...state[action.payload.id],
            ...action.payload,
          },
        };
      }

      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          data: {
            ...state[action.payload.id].data,
            ...action.payload.data,
          },
        },
      };
    }
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.securities;
    default:
      return state;
  }
};

const dataReducersWatchlist = combineReducers({
  status,
  securities,
});

export default dataReducersWatchlist;
