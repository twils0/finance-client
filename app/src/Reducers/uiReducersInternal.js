import { combineReducers } from 'redux';

import { actionTypes, pageBodyNames } from '../Constants/uiConstantsInternal';
import watchlist from './uiReducersWatchlist';
import account from './uiReducersAccount';

export const initialState = {
  utilities: {
    menu: {
      securities: 5, // number of securities in menu
      widthBasis: 600,
      widthBuffer: 60,
      height: 73,
      heightBuffer: 30,
    },
    menuImg: {
      width: 250,
    },
    menuSearchBar: {
      inputLagTime: 200, // time in milliseconds to wait before search
      width: 300,
      height: 45,
      heightBuffer: 10,
    },
    menuSecurity: {
      widthBuffer: 20,
      heightBuffer: 20,
    },
  },
  pageBodies: {
    current: pageBodyNames.WATCHLIST,
    list: [...Object.values(pageBodyNames)],
    [pageBodyNames.WATCHLIST]: { id: pageBodyNames.WATCHLIST },
    [pageBodyNames.ACCOUNT]: { id: pageBodyNames.ACCOUNT },
  },
};

initialState.utilities.menu.mediaWidth = 150
  + 110 // for the searchbar
  + initialState.utilities.menu.widthBasis
  + initialState.utilities.menu.widthBuffer
  + initialState.utilities.menuSearchBar.width
  + initialState.utilities.menu.securities * initialState.utilities.menuSecurity.widthBuffer;

export const utilities = (state = initialState.utilities) => state;

export const pageBodies = (state = initialState.pageBodies, action) => {
  switch (action.type) {
    case actionTypes.SET_CURRENT_PAGE_BODY:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

const uiReducersInternal = combineReducers({
  utilities,
  pageBodies,
  watchlist,
  account,
});

export default uiReducersInternal;
