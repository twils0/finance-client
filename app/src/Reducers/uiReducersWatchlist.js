import { combineReducers } from 'redux';

export const initialState = {
  utilities: {
    pageBody: {
      mediaWidth: 925,
    },
    list: {
      springSettings: { stiffness: 300, damping: 50 },
      width: 300,
      widthBufferLeft: 30,
      widthBufferRight: 15,
      widthBufferScrollBar: 20,
      heightBuffer: 60,
      rowWidthIndent: 10,
      rowHeight: 61,
      rowHeightIndent: 5,
      rowHeightBufferBottom: 10,
      burgerMenuIconWidth: 30,
      burgerMenuIconWidthIndent: 10,
      burgerMenuIconHeight: 27,
    },
    separatorLine: {
      width: 4,
      heightBuffer: 40,
    },
    article: {
      widthBuffer: 60,
      widthIndent: 30,
      mediaWidth: 700,
      heightBuffer: 60,
      heightIndent: 30,
    },
    articleMenu: {
      height: 75,
    },
    articleMenuSecurity: {
      widthBuffer: 40,
      heightBuffer: 20,
    },
    articleGrid: {
      widthBuffer: 20,
      widthIndent: 10,
      heightBuffer: 20,
      heightIndent: 10,
    },
    articleTable: {
      width: 470,
      widthBuffer: 20,
      heightBuffer: 20,
    },
    articleDescription: {
      minWidth: 250,
      widthBuffer: 20,
      heightBuffer: 20,
    },
  },
};

initialState.utilities.list.responsiveHeight = 4 * (initialState.utilities.list.rowHeight + initialState.utilities.list.rowHeightBufferBottom);

initialState.utilities.article.outsideWidthTotal = initialState.utilities.list.width
  + initialState.utilities.list.widthBufferLeft
  + initialState.utilities.list.widthBufferRight
  + initialState.utilities.list.widthBufferScrollBar
  + initialState.utilities.separatorLine.width
  + initialState.utilities.article.widthBuffer;

export const utilities = (state = initialState.utilities) => state;

const uiReducersWatchlist = combineReducers({
  utilities,
});

export default uiReducersWatchlist;
