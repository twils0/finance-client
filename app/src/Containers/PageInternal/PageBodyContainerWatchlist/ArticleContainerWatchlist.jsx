import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestStatusTypes } from '../../../Constants/universalConstants';
import { statusNames } from '../../../Constants/dataConstantsWatchlist';
import requestSecurityData from '../../../Actions/dataThunkWatchlist/requestSecurityData';

import ArticleWatchlist from '../../../Components/PageInternal/PageBodyWatchlist/Article/ArticleWatchlist';

class ArticleContainerWatchlist extends React.Component {
  constructor(props) {
    super(props);

    const { securities } = this.props;
    const { current } = securities;

    if (current && securities[current]) {
      const currentSecurity = securities[current];

      const { handleSecurityData } = this.props;
      const { id, tickerCusip } = currentSecurity;

      console.log('test con sec data 1');

      handleSecurityData({ id, tickerCusip });
    }
  }

  componentWillReceiveProps(nextProps) {
    const { securities } = this.props;
    const newSecurities = nextProps.securities;

    if (securities.current !== newSecurities.current) {
      const { handleSecurityData } = this.props;
      const { current } = newSecurities;

      if (current && newSecurities[current]) {
        const currentSecurity = newSecurities[current];

        const { id, tickerCusip } = currentSecurity;

        handleSecurityData({ id, tickerCusip });
      }
    }
  }

  componentWillUnmount() {
    console.log('test art unmount');
  }

  render() {
    const {
      outsideWidthTotal,
      widthTable,
      widthBuffer,
      widthIndent,
      mediaWidth,
      menuHeight,
      articleMenuHeight,
      heightBuffer,
      heightIndent,
      status,
      securities,
    } = this.props;
    const statusLoading = status[statusNames.SECURITIES].status !== requestStatusTypes.SUCCESS
      || status[statusNames.GET_SECURITY_DATA].status === requestStatusTypes.LOADING;

    return (
      <ArticleWatchlist
        outsideWidthTotal={outsideWidthTotal}
        widthTable={widthTable}
        widthBuffer={widthBuffer}
        widthIndent={widthIndent}
        mediaWidth={mediaWidth}
        menuHeight={menuHeight}
        articleMenuHeight={articleMenuHeight}
        heightBuffer={heightBuffer}
        heightIndent={heightIndent}
        statusLoading={statusLoading}
        securities={securities}
      />
    );
  }
}

ArticleContainerWatchlist.propTypes = {
  outsideWidthTotal: PropTypes.number.isRequired,
  widthTable: PropTypes.number.isRequired,
  widthBuffer: PropTypes.number.isRequired,
  widthIndent: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  menuHeight: PropTypes.number.isRequired,
  articleMenuHeight: PropTypes.number.isRequired,
  heightBuffer: PropTypes.number.isRequired,
  heightIndent: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
  status: PropTypes.object.isRequired,
  handleSecurityData: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  outsideWidthTotal: state.ui.internal.watchlist.utilities.article.outsideWidthTotal,
  widthTable: state.ui.internal.watchlist.utilities.articleTable.width,
  widthBuffer: state.ui.internal.watchlist.utilities.article.widthBuffer,
  widthIndent: state.ui.internal.watchlist.utilities.article.widthIndent,
  mediaWidth: state.ui.internal.watchlist.utilities.pageBody.mediaWidth,
  menuHeight: state.ui.internal.utilities.menu.height,
  articleMenuHeight: state.ui.internal.watchlist.utilities.articleMenu.height,
  heightBuffer: state.ui.internal.watchlist.utilities.article.heightBuffer,
  heightIndent: state.ui.internal.watchlist.utilities.article.heightIndent,
  securities: state.data.watchlist.securities,
  status: state.data.watchlist.status,
});

const mapDispatchToProps = dispatch => ({
  handleSecurityData: payload => dispatch(requestSecurityData(payload)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ArticleContainerWatchlist);
