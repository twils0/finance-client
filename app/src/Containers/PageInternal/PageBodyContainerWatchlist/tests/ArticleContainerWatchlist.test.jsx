import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateWatchlist } from '../../../../Reducers/dataReducersWatchlist';
import { initialState as initialStateUIInternal } from '../../../../Reducers/uiReducersInternal';
import { initialState as initialStateUIWatchlist } from '../../../../Reducers/uiReducersWatchlist';

import requestSecurityData from '../../../../Actions/dataThunkWatchlist/requestSecurityData';

import ArticleContainerWatchlist from '../ArticleContainerWatchlist';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

jest.mock('../../../../Actions/dataThunkWatchlist/requestSecurityData', () => jest.fn());

const shallowComponent = store => shallow(<ArticleContainerWatchlist />, {
  context: {
    store,
  },
}).dive();

const testId0 = 'testId0';
const testTickerCusip0 = 'testTickerCusip0';
const testId1 = 'testId1';
const testTickerCusip1 = 'testTickerCusip1';
const testId2 = 'testId2';
const testTickerCusip2 = 'testTickerCusip2';

const securities = {
  current: testId0,
  list: [testId0, testId1, testId2],
  [testId0]: {
    id: testId0,
    name: 'testName0',
    category: 'testCategory0',
    exchange: 'testExchange0',
    tickerCusip: testTickerCusip0,
  },
  [testId1]: {
    id: testId1,
    name: 'testName1',
    category: 'testCategory1',
    exchange: 'testExchange1',
    tickerCusip: testTickerCusip1,
  },
  [testId2]: {
    id: testId2,
    name: 'testName2',
    category: 'testCategory2',
    exchange: 'testExchange2',
    tickerCusip: testTickerCusip2,
  },
};

describe('Containers', () => {
  describe('PageInternal', () => {
    describe('PageBodyContainerWatchlist', () => {
      describe('ArticleContainerWatchlist', () => {
        afterEach(() => {
          requestSecurityData.mockReset();
        });

        it('shallow renders correctly', async () => {
          const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

          const noDataSecurities = JSON.parse(JSON.stringify({ ...securities }));
          delete noDataSecurities.testId0.data;

          stateBeforeWatchlist.securities = noDataSecurities;

          const store = mockStore({
            data: {
              watchlist: stateBeforeWatchlist,
            },
            ui: {
              internal: {
                ...initialStateUIInternal,
                watchlist: initialStateUIWatchlist,
              },
            },
          });

          const expectedActions = [];

          requestSecurityData.mockReturnValue(() => null);

          const wrapper = shallowComponent(store);

          const actions = store.getActions();

          expect(actions).toEqual(expectedActions);
          expect(wrapper).toMatchSnapshot();
          expect(requestSecurityData).toBeCalledWith({
            id: testId0,
            tickerCusip: testTickerCusip0,
          });
        });

        it('shallow renders correctly, with componentWillReceiveProps', async () => {
          const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

          stateBeforeWatchlist.securities = securities;

          const store = mockStore({
            data: {
              watchlist: stateBeforeWatchlist,
            },
            ui: {
              internal: {
                ...initialStateUIInternal,
                watchlist: initialStateUIWatchlist,
              },
            },
          });

          const nextProps = {
            securities: JSON.parse(JSON.stringify({ ...securities, current: testId1 })),
          };
          delete nextProps.securities.testId1.data;

          const expectedActions = [];

          requestSecurityData.mockReturnValue(() => null);

          const wrapper = shallowComponent(store);

          const instance = wrapper.instance();

          instance.componentWillReceiveProps(nextProps);

          const actions = store.getActions();

          expect(actions).toEqual(expectedActions);
          expect(wrapper).toMatchSnapshot();
          expect(requestSecurityData).toBeCalledWith({
            id: testId1,
            tickerCusip: testTickerCusip1,
          });
        });
      });
    });
  });
});
