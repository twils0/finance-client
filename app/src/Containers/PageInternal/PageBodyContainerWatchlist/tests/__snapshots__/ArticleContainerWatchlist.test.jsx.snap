// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageInternal PageBodyContainerWatchlist ArticleContainerWatchlist shallow renders correctly 1`] = `
<ArticleWatchlist
  articleMenuHeight={75}
  heightBuffer={60}
  heightIndent={30}
  mediaWidth={925}
  menuHeight={73}
  outsideWidthTotal={429}
  securities={
    Object {
      "current": "testId0",
      "list": Array [
        "testId0",
        "testId1",
        "testId2",
      ],
      "testId0": Object {
        "category": "testCategory0",
        "exchange": "testExchange0",
        "id": "testId0",
        "name": "testName0",
        "tickerCusip": "testTickerCusip0",
      },
      "testId1": Object {
        "category": "testCategory1",
        "exchange": "testExchange1",
        "id": "testId1",
        "name": "testName1",
        "tickerCusip": "testTickerCusip1",
      },
      "testId2": Object {
        "category": "testCategory2",
        "exchange": "testExchange2",
        "id": "testId2",
        "name": "testName2",
        "tickerCusip": "testTickerCusip2",
      },
    }
  }
  statusLoading={true}
  widthBuffer={60}
  widthIndent={30}
  widthTable={470}
/>
`;

exports[`Containers PageInternal PageBodyContainerWatchlist ArticleContainerWatchlist shallow renders correctly, with componentWillReceiveProps 1`] = `
<ArticleWatchlist
  articleMenuHeight={75}
  heightBuffer={60}
  heightIndent={30}
  mediaWidth={925}
  menuHeight={73}
  outsideWidthTotal={429}
  securities={
    Object {
      "current": "testId0",
      "list": Array [
        "testId0",
        "testId1",
        "testId2",
      ],
      "testId0": Object {
        "category": "testCategory0",
        "exchange": "testExchange0",
        "id": "testId0",
        "name": "testName0",
        "tickerCusip": "testTickerCusip0",
      },
      "testId1": Object {
        "category": "testCategory1",
        "exchange": "testExchange1",
        "id": "testId1",
        "name": "testName1",
        "tickerCusip": "testTickerCusip1",
      },
      "testId2": Object {
        "category": "testCategory2",
        "exchange": "testExchange2",
        "id": "testId2",
        "name": "testName2",
        "tickerCusip": "testTickerCusip2",
      },
    }
  }
  statusLoading={true}
  widthBuffer={60}
  widthIndent={30}
  widthTable={470}
/>
`;
